Java/Match.java

```diff
@@ -0,0 +1,42 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Example of using regular expressions for finding strings.
+ */
+public final class Match {
+
+    /**
+     * Private constructor prevents class from being instantiated.
+     */
+    private Match() {
+
+    }
+
+    /**
+     * Entry point of the program.
+     *
+     * @param args command-line arguments, not used
+     */
+    public static void main(final String[] args) {
+        String input1 = "My email is someones@mail.com";
+        String input2 = "My email is not@mail";
+        String regularExpression = "[^\\s]+@[^\\s]+\\.[^\\s]+";
+
+        Pattern pattern = Pattern.compile(regularExpression);
+
+        Matcher matcher1 = pattern.matcher(input1);
```

Comment:
To make this more explicit and avoid duplicate code I'd suggest to put all inputs in a collection and then output results with something like
`input: result` or `'input': result`.